src/calculateAverage.js

```diff
@@ -37,7 +37,28 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === undefined) {
+      return this.reduce((sum, item) => item + sum, 0) / this.length;
+    }
+
+    if (options === null) {
+      return undefined;
+    }
+
+    if (options.propertyName) {
+      const value = options.propertyName;
+      const listOfOptions = this.filter(item => item[value]
+        && !isNaN(item[value]));
```

Comment:
Do you really need it?

Reply:
Yes, because I need to check my key value on NaN,

Reply:
You do not really need, NaN will not pass first check `item[value]`

Reply:
![image](https://user-images.githubusercontent.com/50458410/64293438-3f33e680-cf75-11e9-85bb-1db3c445d6f4.png)
Without this expression

Reply:
Ok, got it